[std] Reword "cannot" in notes to not sound like negative permission (#6770)
cplusplus/draft: source/algorithms.tex

where \tcode{is_trivially_copy_constructible_v<T>}
and \tcode{is_trivially_destructible_v<T>} are \tcode{true}.
\begin{note}

> This implies that there is no reliable object identity of arguments
> in user-supplied function objects for such input sequences.

Reviewer: ```suggestion
This implies that the object identity of arguments
in user-supplied function objects is not reliable for such input sequences.
```

Author: Nice, done!
